hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -320,6 +321,11 @@ private <BuilderT extends S3BaseClientBuilder<BuilderT, ClientT>, ClientT> void
       origin = "SDK region chain";
     }
 
+    if (endpointStr != null && endpointStr.endsWith(CENTRAL_ENDPOINT)) {
```

Comment:
instead of this, I think what we should do is on line 307, where it says:

```
    if (region != null) {
      builder.region(region);
    } 
```

do 

```
    if (region != null) {
      builder.region(region);
      if (region == US_EAST_1 && endpointStr != null && endpointStr.endsWith(CENTRAL_ENDPOINT)) {
        builder.crossRegionAccessEnabled(true);
       LOG.debug("Enabling cross region access for endpoint {}", endpointStr);
      }
    } 
```

you will only get into that if block if you haven't set `fs.s3a.endpoint.region` and a region **could** be determined from your `fs.s3a.endpoint.region` , which will happen in this case. As `getS3RegionFromEndpoint` will return US_EAST_1 for s3.amazonaws.com.

Currently we are setting cross region enabled whenever fs.s3a.endpoint = s3.amazonaws.com, even if you know your region and you've set it correctly in fs.s3a.endpoint.region.

Reply:
let's also set the region to US_EAST_2 when enabling cross region. When I was doing this work a few months ago, I found that cross region with US_EAST_1 was showing some weird behaviours..I didn't dive into them at the time. Everything worked as expected with US_EAST_2.

Reply:
Interesting!
You mean to say, with US_EAST_1 cross region, we have some intermittent issues while accessing bucket from other region? Or we always have issues?

Reply:
yeah, I saw issues..403's with the landstat buckets iirc. Could be related to [this](https://github.com/aws/aws-sdk-java/issues/1338#issuecomment-335841213). I don't know if those issues still exist, it could be worth testing with US_EAST_1 to see. It shouldn't make a difference what we set though, if you haven't got the correct region SDK will make a call to figure out the correct region and then cache it

Reply:
does this look good for source changes https://github.com/apache/hadoop/pull/6479/commits/05225bb74669d5261a9cc376c1dd974c9f81643d ?

Reply:
landsat bucket seems to have been closed off; we will need to move off it -but the replacement must be something other than us-east so it stresses more of the system